analytics.go

```diff
@@ -250,25 +251,25 @@ func (c *client) upload(b []byte) error {
 	}
 
 	defer res.Body.Close()
-	c.report(res)
-
-	return nil
+	return c.report(res)
 }
 
 // Report on response body.
-func (c *client) report(res *http.Response) {
+func (c *client) report(res *http.Response) (err error) {
+	var body []byte
+
 	if res.StatusCode < 400 {
```

Comment:
Do you know why this condition was expressed this way? It feels to me that anything but a 200 should be considered an error (the [doc](https://segment.com/docs/libraries/http/#errors) says the API always returns 200).